webrender/src/prim_store.rs

```diff
@@ -677,23 +662,55 @@ impl PrimitiveStore {
                                    resource_cache: &mut ResourceCache,
                                    frame_id: FrameId,
                                    device_pixel_ratio: f32,
+                                   dummy_mask_cache_item: &TextureCacheItem,
                                    auxiliary_lists: &AuxiliaryLists) -> bool {
         let metadata = &mut self.cpu_metadata[prim_index.0];
         debug_assert!(metadata.need_to_build_cache);
         metadata.need_to_build_cache = false;
 
-        if let Some(mask_key) = metadata.mask_image {
-            let tex_cache = resource_cache.get_image(mask_key, ImageRendering::Auto, frame_id);
+        let clip_data_mask = match metadata.clip_info.as_ref() {
```

Comment:
I don't think we handled the zero radius anywhere except this piece:
```diff
             if scrollbar_prim.border_radius == 0.0 {
-                self.prim_store.set_complex_clip(scrollbar_prim.prim_index, None);
+                self.prim_store.unset_complex_clip(scrollbar_prim.prim_index)
             } else {
```
The general case of no `ComplexClipRegion` given is handled below, here:
```rust
let clip_data = match clips.len() {
                    0 if clip_region.image_mask.is_none() => None,
```
So I don't think this PR changes the logic in a way that makes the shaders more expensive.